packages/react/src/components/Dialog/DialogFooter.tsx

```diff
@@ -0,0 +1,53 @@
+import * as React from 'react'
+import * as customPropTypes from '@stardust-ui/react-proptypes'
+
+import {
+  createShorthandFactory,
+  UIComponent,
+  UIComponentProps,
+  ChildrenComponentProps,
+  ContentComponentProps,
+  commonPropTypes,
+  ShorthandFactory,
+  childrenExist,
+} from '../../lib'
+
+import { ShorthandValue, WithAsProp, withSafeTypeForAs } from '../../types'
+import { BoxProps } from '../Box/Box'
+
+export interface DialogFooterProps
+  extends UIComponentProps,
+    ChildrenComponentProps,
+    ContentComponentProps {
+  /** Dialog actions */
+  actions: ShorthandValue<BoxProps>
+}
+
+class DialogFooter extends UIComponent<WithAsProp<DialogFooterProps>> {
+  static create: ShorthandFactory<DialogFooterProps>
+
+  static displayName = 'DialogFooter'
+  static className = 'ui-dialog__footer'
+
+  static propTypes = {
+    ...commonPropTypes.createCommon(),
+    actions: customPropTypes.itemShorthand,
```

Comment:
```suggestion
```

https://github.com/stardust-ui/react/pull/2005/files#r333918536
In current implementation it's not shorthand, so I don't see any reasons to keep it.